### PR TITLE
Add widget

### PR DESCRIPTION
**Pull request by csdodd**
Allows user to add a widget to their home screen. It just triggers the same service every 4 hours.

4 hours is just a number I pulled out of nowhere to get a fine balance between usability and performance.

screen shot 2017-10-27 at 13 34 12

resize

### CODE REVIEW

**Comment by martintoften** [comment]
How does widgets work with the background work limitations in Oreo?

File: app/src/main/java/com/bakkenbaeck/sol/widget/DashboardWidget.java [hunk 1]
```diff
@@ -60,13 +60,20 @@ public void onReceive(final Context context, final Intent intent) {
...
             setSunView(context, views, uvd);
             setLocationPermissionMessage(context, views, uvd);
             makeWidgetClickable(context, views);
             appWidgetManager.updateAppWidget(appWidgetManager.getAppWidgetIds(componentName), views);
         }
 
+        private void setBackground(final RemoteViews views,
+                                   final UserVisibleData uvd) {
+            final int color = uvd.getCurrentPhase().getBackgroundColor();
+            views.setInt(R.id.widget_container, "setBackgroundResource", color);
```
Comment (martintoften) [inline app/src/main/java/com/bakkenbaeck/sol/widget/DashboardWidget.java]: Can this be a field variable instead? Easier to read

**Comment by csdodd** [comment]
Actually, I'm not how it performs on Oreo, but I think that should be a seperate PR anyway - to avoid this one growing more.

**Comment by csdodd** [comment]
It's not an issue so long as we target SDK 25. Supporting 26 isn't easy and I don't think we should put the work in yet (as this is just a hobby project). Maybe later on the support lib for this will be better making it quicker to implement.

**Pull request merged**